project_5/main.py

```diff
@@ -0,0 +1,149 @@
+# import libraries
+import os
+import pandas as pd
+import numpy as np
+import gspread
+import logging
+from dotenv import load_dotenv
+import mysql.connector as sql
+import warnings
+warnings.filterwarnings('ignore')
+
+logging.basicConfig(level=logging.INFO)
+
+# connect database
+def connect_db(DATABASE: str, DB_USER: str, DB_PASSWORD: str, DB_HOST: str) -> sql.connection:
+    """
+    functions creates a connection to the database
+    """
+
+    try:
+        connection = sql.connect(
+            database=DATABASE,
+            user=DB_USER,
+            host=DB_HOST,
+            password=DB_PASSWORD,
+        )
+        logging.info("opening connection to database")
+    except:
+        raise Exception("Cant connect to database")
+        
+    return connection
+
+# create table
+def create_table(connection: sql.connection) -> None:
+    """
+    functions creates a table in the specified database if doesnt exist
+    """
+    # query to create table in the database
+    query = """ 
+            CREATE TABLE housing_dataset(
+            serial_no INT AUTO_INCREMENT,
+            ID INT NOT NULL,
+            loc VARCHAR(20) DEFAULT NULL,
+            title VARCHAR(20) DEFAULT NULL,
+            bedroom INT DEFAULT NULL,
+            bathroom INT DEFAULT NULL,
+            parking_space INT DEFAULT NULL,
+            price FLOAT DEFAULT NULL,
+            created_at TIMESTAMP NULL DEFAULT CURRENT_TIMESTAMP,
+            PRIMARY KEY (serial_no)
+    );
+    """
+
+    # execute query to create table
+    try:
+        with connection.cursor() as cursor:
+            cursor.execute(query)
+
+            connection.commit()
+            logging.info("table created successfully")
+    except:
+        logging.info("table already exists")
+    
+    return None
```

Comment:
This function returns `None` regardless of this, so this might be redundant

Reply:
fixed